src/javascript/Create/CreateNewContentAction/createNewContent.action.jsx

```diff
@@ -1,83 +1,83 @@
-import React from 'react';
+import React, {useContext} from 'react';
 import {CreateNewContentDialog} from './CreateNewContentDialog';
-import {componentRendererAction, composeActions} from '@jahia/react-material';
-import {reduxAction} from '../../actions/redux.action';
 import {Constants} from '~/ContentEditor.constants';
-import {cmGoto} from '~/JContent.redux-actions';
-import {withApolloAction} from './withApolloAction';
-import {from} from 'rxjs';
-import {share} from 'rxjs/operators';
-import {transformNodeTypesToActions, getCreatableNodetypes} from './createNewContent.utits';
+import {transformNodeTypesToActions, useCreatableNodetypes} from './createNewContent.utits';
+import {useSelector} from 'react-redux';
+import {useNodeInfo} from '@jahia/data-helper';
+import {ComponentRendererContext} from '@jahia/ui-extender';
+import * as PropTypes from 'prop-types';
+import {useContentEditorHistory} from '~/ContentEditorHistory';
 
-// Should continue to use redux here, because this action is used to replace JContent create action, so it's only displayed in JContent
-const mapDispatchToProps = dispatch => ({
-    gotToCE: gotoParams => dispatch(cmGoto(gotoParams))
-});
+const onClick = (uuid, language, context, redirect, componentRenderer) => {
+    if (context.openEditor) {
+        redirect({mode: Constants.routes.baseCreateRoute, language, uuid, rest: encodeURI(context.nodeTypes[0])});
+    } else {
+        const closeDialog = () => {
+            componentRenderer.destroy('CreateNewContentDialog');
+        };
 
-const mapStateToProps = state => {
-    return {
-        uilang: state.uilang,
-        language: state.language,
-        siteKey: state.site
-    };
+        componentRenderer.render(
```

Comment:
We can go deeper and render the modal directly in the component 🤔 
But as we don't have modal in moonstone yet, I'm to keep it like this.

Reply:
thanks for the advice!